### PR TITLE
added color to selected drawer item

### PR DESCRIPTION
**Pull request by getsanjeev**

### CODE REVIEW

**Comment by getsanjeev** [comment]
@grzesiek2010 I have made suitable changes to get the second part done. Its working on all the activities except for 'Get Blank Forms'. Actually in all other cases, I am getting the 'mSelectedSortingOrder' as the expected one. But in this (get blank form) I am getting null. I have put a log to check it.

MY approach is : 
1. while the list view is being created I am giving the textview of mSelectedSortingOrder a color(blue). 
2. When an item is clicked I am changing the color of previously selected item to normal(default grey) and changing this one to blue.

Its working except for what I mentioned above. Please have a look at it.

**Comment by grzesiek2010** [comment]
@getsanjeev 
It's null because you try to use that before we call `restoreSelectedSortingOrder()` method.
My proposal is to add a new method in [AppListaActivity](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java) and [AppListFragment](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java). This method should look like:

> protected int getSelectedSortingOrder() {
        if (mSelectedSortingOrder == null) {
            restoreSelectedSortingOrder();
        }
        return mSelectedSortingOrder;
    }

then you can make mSelectedSortingOrder private (in both classes) and use this new method instead.

File: collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java [hunk 1]
```diff
@@ -202,6 +208,9 @@ public View getView(int position, View convertView, ViewGroup parent) {
         mDrawerList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
             @Override
             public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                Log.e("selected ",mSelectedSortingOrder+"");
+                parent.getChildAt(mSelectedSortingOrder).setBackgroundColor(Color.parseColor("#DDDDDD"));
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: Please use Color.TRANSPARENT instead.

File: collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java [hunk 2]
```diff
@@ -202,6 +208,9 @@ public View getView(int position, View convertView, ViewGroup parent) {
...
+                view.setBackgroundColor(Color.parseColor("#2196F3"));
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: Please move this value to [colors.xml](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/res/values/colors.xml)

File: collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java [hunk 3]
```diff
@@ -193,6 +195,10 @@ private void setupDrawerItems() {
             @Override
             public View getView(int position, View convertView, ViewGroup parent) {
                 TextView textView = (TextView) super.getView(position, convertView, parent);
+                Log.e("selected ",mSelectedSortingOrder+"");
+                if(position == mSelectedSortingOrder) {
+                    textView.setBackgroundColor(Color.parseColor("#2196F3"));
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: Please move this value to [colors.xml](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/res/values/colors.xml)

File: collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java [hunk 4]
```diff
@@ -193,6 +195,10 @@ private void setupDrawerItems() {
             @Override
             public View getView(int position, View convertView, ViewGroup parent) {
                 TextView textView = (TextView) super.getView(position, convertView, parent);
+                Log.e("selected ",mSelectedSortingOrder+"");
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: We don't need that especially we are going to use Timber as I know.

File: collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java [hunk 5]
```diff
@@ -202,6 +208,9 @@ public View getView(int position, View convertView, ViewGroup parent) {
         mDrawerList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
             @Override
             public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                Log.e("selected ",mSelectedSortingOrder+"");
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: We don't need that especially we are going to use Timber as I know.

File: collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java [hunk 6]
```diff
@@ -172,6 +173,9 @@ private void setupDrawerItems() {
             @Override
             public View getView(int position, View convertView, ViewGroup parent) {
                 TextView textView = (TextView) super.getView(position, convertView, parent);
+                if(position == mSelectedSortingOrder) {
+                    textView.setBackgroundColor(Color.parseColor("#2196F3"));
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java]: Please move this value to colors.xml

File: collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java [hunk 7]
```diff
@@ -180,6 +184,8 @@ public View getView(int position, View convertView, ViewGroup parent) {
         mDrawerList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
             @Override
             public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                parent.getChildAt(mSelectedSortingOrder).setBackgroundColor(Color.parseColor("#DDDDDD"));
+                view.setBackgroundColor(Color.parseColor("#2196F3"));
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java]: Please move this value to colors.xml

File: collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java [hunk 8]
```diff
@@ -180,6 +184,8 @@ public View getView(int position, View convertView, ViewGroup parent) {
         mDrawerList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
             @Override
             public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                parent.getChildAt(mSelectedSortingOrder).setBackgroundColor(Color.parseColor("#DDDDDD"));
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java]: Please use Color.TRANSPARENT instead.

Comment (getsanjeev) [reply collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java]: okay. I will change it.

File: collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java (hunk 5 above)
Comment (getsanjeev) [reply collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java]: certainly will remove all the logs.

**Comment by getsanjeev** [comment]
@lognaturel Please review.

**Comment by grzesiek2010** [comment]
@getsanjeev I don'tlike the way you use getSelectedSortingOrder() please do that as I described above:

> make mSelectedSortingOrder private (in both classes) and use this new method instead

then you don't need to call this method in  onResume/onViewCreated like now.

**Comment by getsanjeev** [comment]
@grzesiek2010 Sorry I do not get your point exactly. It would be helpful if you commit the changes. It would be great to learn your approach.

**Comment by grzesiek2010** [comment]
@getsanjeev I think you are able to do that and you should as it's your part of work. So:
you should to make mSelectedSortingOrder private (now it's protected)
https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java#L74 and https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java#L66

mSelectedSortingOrder is protected because we use this field in other classes:
https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/activities/FormListActivity.java#L12
https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/activities/InstanceListActivity.java#L14
https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/fragments/FormListFragment.java#L26
https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/fragments/InstanceListFragment.java#L27
So just replace private with protected asI mentioned and use getSelectedSortingOrder() i above (4) listed classes.

**Comment by getsanjeev** [comment]
@grzesiek2010 Thanks for being so helpful. I think I got it.

**Comment by getsanjeev** [comment]
@grzesiek2010 I have made the changes you asked for. But still I get the null pointer exception.
The error is simply because we have not initialised mSelectedSortingOrder variable before actually using it in get view(). 
In all the activities we initialise it by calling getSortingOrder() of either FormListActivity or InstanceListActivity in getCursor() function. But In 'Get Blank Form' its different.
Here we are not able to call this function, So we must do it somewhere in activity. How do you want me to do it?

**Comment by grzesiek2010** [comment]
@getsanjeev Why aren't you able to use getSelectedSortingOrder() method in FormDownloadList activity?

**Comment by getsanjeev** [comment]
Well I meant that we havenot called this function yet. I must call it explicitly in activity to initialise it.
So where do you want me to add it.

**Comment by getsanjeev** [comment]
check #896

**Comment by grzesiek2010** [comment]
@getsanjeev just call restoreSelectedSortingOrder() in ApplistActivity#onResume() before other setupSomething methods.

**Comment by getsanjeev** [comment]
Done.

**Pull request closed without merging**